src/sentry/sentry_metrics/consumers/indexer/parallel.py

```diff
@@ -112,8 +111,9 @@ def create_with_partitions(
         commit: Callable[[Mapping[Partition, Position]], None],
         partitions: Mapping[Partition, int],
     ) -> ProcessingStrategy[KafkaPayload]:
+        message_processor = MessageProcessor(self.__config.use_case_id)
         parallel_strategy = ParallelTransformStep(
-            partial(process_messages, self.__config.use_case_id),
+            message_processor.process_messages,
```

Comment:
nit: Please just initialize the processor here. You do not need the `message_processor` variable.